### PR TITLE
Implement network settings UI.

### LINKED ISSUE #423: Network Settings
As a user I want to configure my network settings so that I change the network my gateway is connected to

### LINKED ISSUE #895: WiFi Access Point
As a user I want configure my gateway to support both station and access point mode so that I can connect all my edge device/sensors to gateway (Raspberry Pi) directly.

### LINKED ISSUE #1674: Current IP Address
As a user I would like to find out the current IP address of my gateway if it was automatically assigned.

- When connected as a DHCP client via  WiFi network
- When connected as a DHCP client via Ethernet
- Public IP when connected to the Internet via a WAN port

### PR DESCRIPTION
**Pull request by mrstegeman**
Fixes #423
Fixes #895
Fixes #1674

### CODE REVIEW

File: static/css/settings.css [hunk 1]
```diff
@@ -116,9 +116,8 @@
   background-image: url('../optimized-images/domain-icon.png');
 }
 
-
-#domain-settings-mozilla-link {
```
[the remark this answers is not in the document]
Comment (mrstegeman) [reply static/css/settings.css]: Good catch, got it.

**Comment by mrstegeman** [comment]
> On my pi which is connected via wifi without an ethernet cable plugged in the PUT request to configure the ethernet settings takes 64 seconds to complete. The wifi config takes less time (~3 seconds) but during both it seems like the entire process is blocked. There's also no visual feedback that the click on "Done" was recognized and submitted so the first time I tried the ethernet config and it stalled out I followed up by clicking the button five more times before realizing that spam-clicking never helps, no matter how good it feels. The pi performance issues might just be that my local pi is buckling under the strain of one and a half voice assistants and a ton of other random development cruft.

Yeah, I noticed that network config takes a while on the Pi, which is unfortunate. Maybe I'll disable the button after it's clicked and throw up a message that it may take a minute.

> On mac/windows this should have a "your platform is unsupported" message in network settings instead of a blank page and a console.log

Good call. I can add that.

**Comment by mrstegeman** [comment]
@hobinjk I addressed all of the issues you brought up.

**Comment by benfrancis** [comment]
@mrstegeman Any advice on how to test the Raspbian and OpenWrt versions of this feature?

**Comment by mrstegeman** [comment]
@benfrancis For testing the UI bits, I just added a `darwin` case here: https://github.com/mozilla-iot/gateway/blob/6cb6c6a28d496093011d6817421fb833f2c82d33/static/js/settings.js#L1007-L1015

That way, you can test both the client and the router UI. If you want to test the actual network functionality, you'll need to clone my branch on a Pi or on the Turris and run it as usual.

**Comment by mrstegeman** [comment]
> I've only tested the front end on my PC so haven't seen it fully in action yet but just have a few small comments:
> 
> * background-color of dropdowns should match background-color of buttons (#597285)
> * In Manual mode of Ethernet Settings and WAN Settings I suggest setting network mask to 255.255.255.0 by default
> * For the disabled state of buttons, can we set opacity to make it translucent rather than setting a lighter background-color?

Done.

> I wasn't able to see the Wi-Fi (client) network list in the UI but I'm guessing it looks much the same as first time setup.

Yes, essentially the same but with the added "Connected"/"Disconnected" labels.

> I noticed that current IP address and Wi-Fi network name are not displayed on any of the network options, were you planning on implementing that separately in #1674?

Nope, I just forgot. Added that now.

> Also I guess we can add the in-UI help separately in #677.

Yes, that was my plan.

**Comment by mrstegeman** [comment]
@benfrancis @hobinjk Any further issues?

**Comment by hobinjk** [comment]
:+1:

**Pull request merged**